### ISSUE TITLE
Deliverers cannot be identified based on extracted data from the document

### ISSUE THREAD

**Issue by astorije** [w3c/specberus#130]
At the moment, Specberus extracts from the documents the name of the WG as it is displayed, and the underlying link, which is the homepage.

Unfortunately, none of these are enough to properly identify a Working Group without heuristics:
- The displayed name can change from one document to another, like *[Web Applications WG](http://www.w3.org/TR/2015/REC-IndexedDB-20150108/)*, *[Web Applications Working Group](http://www.w3.org/TR/2014/WD-DOM-Level-3-Events-20140925/)*, *[Web Applications (WebApps) Working Group](http://www.w3.org/TR/2013/WD-appmanifest-20131217/)*, ...
- The homepage can match to multiple Working Groups in W3C database

I have no good solution to suggest for this, and I don't think a solution. One way (that I dislike) could be to add the ID of the group (or any other unique identifier) to the document similarly to the editors.
Another one (that I like) would be to make the homepage unique. We currently have only 100 link duplicates in the corresponding table, 8 duplicates for homepages that contain `w3.org`.

**Comment by tripu** [comment]
I think the best approach would be to add more metadata to specs, in this case in the form of a new `data-deliverer-id` attribute. It makes perfect sense to expect specs to be machine-readable, if we want them to be ultimately *machine-publishable*.

I suspected this would be the conundrum as soon as we starting adding *some* metadata to specs, and I'm glad that things are pushing us towards more semantic specs.

That being said, I await your comments, and a decision from the group, before tackling this one.

**Comment by deniak** [comment]
+1 on @tripu 's comment. If we are asking the editors to add their id in the document, adding the WG's id shouldn't be too difficult and gives us a good way to identify the deliverers

**Comment by astorije** [comment]
-1 on the WG ID suggestion, as said on my previous message. I agree we need all possible metadata, but the ID makes things obscure and confusing. Do we have another piece of information to identify the group that we can make use of?
I won't fight over the ID suggestion though, we don't have time for this.

I would like to have feedback from @plehegar, @darobin and @dontcallmedom however.

**Comment by dontcallmedom** [comment]
The current checker relies on the WG homepage — I'm not sure why this needs to change now; we can look to a more reliable system overtime, but I don't think that should be a blocking factor for specberus in the short term.

**Comment by astorije** [comment]
@dontcallmedom, unfortunately it is a blocking factor right now. The current checker and, therefore, the current publication system, cannot identify properly a WG from a document. Mainly 2 things can happen (but I might overlook other issues):

- The WG homepage found corresponds to several WG (that is the case for the HTML WG, which shares its homepage with the *HTML WG Editors* and *HTML Weekly*)
- The SOTD mentions multiple WGs, for example: "This document is published by BlaBla WG and was reviewed by the FooBar WG"

In both cases, only 1 WG is publishing and it's a manual task to identify the false positives.
In short, we need a way to identify the WG accurately. Solving the homepage duplicates would be doable IMO but it only solves the first problem...

**Comment by dontcallmedom** [comment]
the link to the IPP status page contains the wgid of the relevant group

**Comment by astorije** [comment]
Wow, hack alert.
Indeed, I didn't see that, so in that case we can certainly add the ID around the WG, I guess...
However, extracting it to construct the group ID seems like a bad idea to me, because it is obscure and will make false positives and errors hard to figure out.

**Comment by astorije** [comment]
@plehegar and I are currently discussing, and enforcing the displayed name of the WG is actually required, otherwise one could publish a doc saying "published by the FooBar WG" as long as all the links are correct.

**Comment by tripu** [comment]
TODO @tripu: change the way we extract that metatada (ID within IPP link, instead of new data attribute; for the time being).
